data/tutorials/getting-started/gs_02_fst_program.md

```diff
@@ -0,0 +1,356 @@
+
+The project is stored in a directory named `hello`. The `tree` command lists the files and directories created. It might be necessary to install `tree` if you don't see the following. Through Homebrew, for example, run `brew install tree`.
+
+Note: If you get an error in Homebrew from this in an Apple silicon macOS, it's likely an issue with the architecture switch from Intel to ARM. Please refer to the [ARM64 Fix](FIXME) to remedy the ARM64 error.
+```shell
+$ cd hello
+$ tree
+.
+├── bin
+│   ├── dune
+│   └── main.ml
+├── _build
+│   └── log
+├── dune-project
+├── hello.opam
+├── lib
+│   └── dune
+└── test
+    ├── dune
+    └── hello.ml
+
+4 directories, 8 files
+```
+
+OCaml source files have `.ml` extension, which stands for “Meta Language.” Meta Language (ML) is the ancestor of OCaml, this is also what the “ml” stands for in “OCaml.” Here is the content of the `bin/main.ml` file:
+```ocaml
+let () = print_endline "Hello, World!"
+```
+
+The project-wide metadata is available in the `dune-project` file. Each folder containing source files that need to be built must contain a `dune` file explaining how.
+
+This builds the project:
+```shell
+$ dune build
+Entering directory '/home/cuihtlauac/caml/ocaml.org'
+```
+
+This launches the executable it creates:
+```shell
+$ dune exec hello
+Entering directory '/home/cuihtlauac/caml/ocaml.org'
+Hello, World!
+```
+
+Let's see what happens when we edit the `bin/main.ml` file directly. Navigate into it and replace the world `World` with your first name. Recompile the project with `dune build` as before, and then launch it again with `dune exec hello`.
+
+Voilà! You've just written your first OCaml program.
+
+In the rest of this tutorial, we will make more changes to this project in order to illustrate OCaml's tooling.
+
+## Why Isn't There a Main Function?
+
+Although `bin/main.ml`'s name accurately suggests it contains the application entry point into the executable binary produced by the project, it does not contain a dedicated `main` function. Even that file name is irrelevant, it could be anything. This is because OCaml is equally behaves as an interpreted and a compiled language. The toplevel reads input, type checks it, compiles it and executes it, triggering computation and side effects. Execution of the binaries generated by the compiler behaves as if interpretation was taking place, the only difference should be speed.
+
+That's why double semicolumns aren't needed in source files like they are in the toplevel. Statements are just processed in order from top to bottom, each triggering the side effects it may have. Definitions are added to the environment. Values resulting from nameless expressions are ignored. Side effects from all those will take place in the same order. That's OCaml main.
```

Comment:
```suggestion
That's why double semicolons aren't needed in source files like they are in the toplevel. Statements are just processed in order from top to bottom, each triggering the side effects it may have. Definitions are added to the environment. Values resulting from nameless expressions are ignored. Side effects from all those will take place in the same order. That's OCaml main.
```